### PR TITLE
Update envoy version to v1.23, and contour to v1.22

### PR DESCRIPTION
**Pull request by nshankar13**
Signed-off-by: nshankar13 <nshankar@microsoft.com>

**Description**: Update envoy version to v1.23.1. To make this compatible with contour, we will also need to bump contour to v1.22.1 

Checked the release notes of envoy v1.23.1, nothing here seems to impact OSM. https://www.envoyproxy.io/docs/envoy/v1.23.1/version_history/v1.23/v1.23.1

Contour v1.22.1 changelog also does not impact OSM: https://github.com/projectcontour/contour/releases/v1.22.1

**Testing done**: Tested OSM installation and sidecar injection on ARM64 machine and ran through contour ingress demo successfully. 

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Demo                       | [ ] |
| Documentation              | [ ] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [X] |
| Tests                      | [ ] |
| Upgrade                    | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

2. Is this a breaking change?

3. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)?

### CODE REVIEW

**Comment by jsturtevant** [comment]
@nshankar13 anything holding this back? Seems like we should get this in early in the release cycle for 1.3 to get test coverage.  Could you do the rebase required?

**Comment by nshankar13** [comment]
@jsturtevant yep we're blocked because the ARM image for envoy-distroless v1.23.0 doesn't have the correct binary: https://github.com/openservicemesh/osm/issues/4716#issuecomment-1203111096. I was instead thinking we should just wait for envoy-distroless:v1.23.1 so that way we could also have the multi-arch issue sorted out. 

However, if we need envoy v1.23.0 in soon I can go ahead and restart this PR, and wait until v1.23.1 and do that as a separate update for the ARM64 testing.

**Comment by jsturtevant** [comment]
do we currently support ARM?  If we don't then wouldn't block upgrading early in the cycle.

**Comment by nshankar13** [comment]
Good point, we don't. I'll get this PR restarted again.

File: charts/osm/values.yaml
Comment (shashankram) [inline charts/osm/values.yaml:53]:
```suggestion
  # -- NOTE: This should point to digest of the manifest that points to both the AMD and ARM images rather than one of the two
```

**Review by shashankram — COMMENTED** [review]
Could you update the PR description to indicate the testing done. Looks good otherwise.

**Comment by odidev** [comment]
> However, if we need envoy v1.23.0 in soon I can go ahead and restart this PR, and wait until v1.23.1 and do that as a separate update for the ARM64 testing.

@nshankar13  envoy-distroless v1.23.1 is released in [dockerhub](https://hub.docker.com/r/envoyproxy/envoy-distroless/tags)

**Comment by nshankar13** [comment]
Thanks @odidev, will update to Envoy v1.23.1 and test again.

**Comment by nshankar13** [comment]
Actually @odidev, we will also need to wait for contour to support envoy v1.23.1 as well, putting this on hold until that's available: https://github.com/projectcontour/contour/releases/

**Comment by keithmattix** [comment]
FWIW Looks like that new release is coming soon: https://github.com/projectcontour/contour/pull/4693

**Comment by keithmattix** [comment]
@nshankar13  New contour version was just released!

File: charts/osm/values.yaml
Comment (steeling) [inline charts/osm/values.yaml:623]: This value looks like it's present in values.schema.json

**Review by steeling — APPROVED** [review]

**Review by keithmattix — APPROVED** [review]

**Comment by keithmattix** [comment]
Approving pending green CI

**Comment by nshankar13** [comment]
Still need to test Contour demo and sidecar injection, hold off on merging until then.

**Comment by nshankar13** [comment]
@keithmattix @shashankram Tested sidecar injection on ARM64 and ran the contour demo successfully, and updated PR description.

**Review by shashankram — APPROVED** [review]

**Pull request merged**
